### PR TITLE
Show qdtype info in musical score diagrams

### PR DESCRIPTION
**Pull request by tanujkhattar**
Fixes https://github.com/quantumlib/Qualtran/issues/854

Currently based on top of https://github.com/quantumlib/Qualtran/pull/856 since I imagine this PR can involve more discussion than https://github.com/quantumlib/Qualtran/pull/856

Can also default to show only the size instead of `str(dtype)` if people think this is cluttering the output too much.

### CODE REVIEW

**Review by mpharrigan — APPROVED** [review]
Easy way to get at least the dtypes of the dangling wires. 

Nit: I think of bracketed numbers as indexing in to something rather than the size of something. My preference would be to use `/` 

`x / QInt(5)`. 

or maybe parentheses

File: qualtran/drawing/musical_score.py [hunk 1]
```diff
@@ -476,7 +477,7 @@ def _soq_to_symb(soq: Soquet) -> WireSymbol:
 
     # Use text (with no box) for dangling register identifiers.
     if isinstance(soq.binst, DanglingT):
-        return Text(soq.pretty())
+        return Text(soq.pretty() + f'[{soq.reg.dtype!s}]', fontsize=8)
```
Comment (mpharrigan) [inline qualtran/drawing/musical_score.py:480]: is !s necessary? isn't that the default?

Comment (tanujkhattar) [reply qualtran/drawing/musical_score.py:480]: you are right, it's not needed. Removed.

**Comment by tanujkhattar** [comment]
Change to `/` instead of `[]` as suggested, so it'll now view as `x / QInt(5)`

**Pull request merged**